### PR TITLE
fix: cease many model fetch api calls in checkpoint tab

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1950]

## Test Plan
- Check if model fetch api is only called once in init phase
- Check if checkpoint `delete` and `register` still works

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1950]: https://hpe-aiatscale.atlassian.net/browse/WEB-1950?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/CheckpointModalTrigger.tsx
Comment (keita-determined) [inline webui/react/src/components/CheckpointModalTrigger.tsx:1]: ref: https://react.dev/learn/you-might-not-need-an-effect#initializing-the-application

**Review by gt2345 — COMMENTED** [review]
I'm not sure about this approach, if we've done everything else correct, and `fetchModels` run twice, then adding a top-level variable makes sense

**Comment by keita-determined** [comment]
> I'm not sure about this approach, if we've done everything else correct, and `fetchModels` run twice, then adding a top-level variable makes sense

i used [this approach](https://react.dev/learn/you-might-not-need-an-effect#initializing-the-application). i think the api is already called once or twice for each `register/delete` component, but there are several `register/delete` components in a checkpoint page. Thats why api is called several times ((1 or 2 api calls) * checkpoint counts in a page).

**Comment by keita-determined** [comment]
im the wrong one probably. its not working right rn. let me investigate it more

**Comment by keita-determined** [comment]
@gt2345 The original issue is that `src/components/CheckpointModalTrigger.tsx` calls `model fetch`, so if there are 100 checkpoints in a page, the api is called at least 100 times. I moved the fetching api logic into the top level component.

**Review by gt2345 — COMMENTED** [review]
Works as expected. 
Do you think we should put models in a store?

**Comment by keita-determined** [comment]
> Works as expected. Do you think we should put models in a store?

for now, i dont think so because the usage of model is not as much as others.

File: webui/react/src/hooks/useFetchModels.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        type: ErrorType.Api,
+      });
+    }
+  }, [canceler.signal]);
+
+  useEffect(() => {
+    fetchModels();
+  }, [fetchModels]);
+
+  return models;
```
Comment (ashtonG) [inline webui/react/src/hooks/useFetchModels.tsx:47]: this (async call that only runs when dependencies change) is a good use case for `useAsync`. consider instead:
```suggestion
  return useAsync(async (canceler) => {
    try {
      const response = await getModels(
        {
          archived: false,
          orderBy: 'ORDER_BY_DESC',
          sortBy: validateDetApiEnum(
            V1GetModelsREquestSortBy,
            V1GetModelsRequestSortBy.LASTUPDATEDTIME,
          ),
        },
        { signal: canceler.signal },
      );
      return response.models;
    } catch (e) {
      handleError(e, {
        publicSubject: 'Unable to fetch models.',
        silent: true,
        type: ErrorType.Api,
      });
      return NotLoaded
    }
  }, [])
```

Comment (keita-determined) [reply webui/react/src/hooks/useFetchModels.tsx]: done!

**Review by ashtonG — APPROVED** [review]

**Pull request merged**